book/src/powdr_crate.md

```diff
@@ -0,0 +1,72 @@
+# Using powdr as a library
+
+Besides the CLI, powdr can also be used as a Rust library.
+The [powdr crate](https://github.com/powdr-labs/powdr/tree/main/powdr)
+exposes internal crates and data structures needed to compile code,
+generate witnesses, and compute proofs.
+
+Add `powdr` to your crate's dependencies:
+
+```toml
+[dependencies]
+powdr = { git = "https://github.com/powdr-labs/powdr", branch = "main" }
+```
+
+The following Rust code has the same workflow as our previous "Hello World" example.
+The full project can be found [here](https://github.com/powdr-labs/powdr-hello-world).
+
+```rust
```

Comment:
Would there be a way to test this file? How about adding it as an example to the crate?

Reply:
Yea maybe I can actually add this as a test, and then include it from the file into the docs. I think some asm files are like this